src/plugins/explore/public/application/utils/state_management/actions/set_dataset/set_dataset.ts

```diff
@@ -58,6 +60,14 @@ export const setDatasetActionCreator = (
     dispatch(setPromptModeIsAvailable(newPromptModeIsAvailable));
   }
 
+  const newSummaryAgentIsAvailable = await getSummaryAgentIsAvailable(
+    services,
+    query.dataset?.dataSource?.id!
+  );
+  if (newSummaryAgentIsAvailable !== summaryAgentIsAvailable) {
+    dispatch(setSummaryAgentIsAvailable(newSummaryAgentIsAvailable));
+  }
+
```

Comment:
i'm pretty sure the unit tests for this file would need to be updated for this change